receiver/hostmetricsreceiver/hostmetrics_receiver.go

```diff
@@ -160,3 +238,14 @@ func (hmr *receiver) closeScrapers(ctx context.Context) error {
 
 	return nil
 }
+
+func (hmr *receiver) allScrapers() []internal.BaseScraper {
+	allScrapers := make([]internal.BaseScraper, len(hmr.hostMetricScrapers)+len(hmr.resourceMetricScrapers))
+	for i, hostMetricScraper := range hmr.hostMetricScrapers {
+		allScrapers[i] = hostMetricScraper
+	}
+	for i, resourceMetricScraper := range hmr.resourceMetricScrapers {
+		allScrapers[len(hmr.hostMetricScrapers)+i] = resourceMetricScraper
```

Comment:
nit: extract `len(hmr.hostMetricScrapers)` to a variable outside of the loop and name it something like `offset`